### ISSUE TITLE
Setting GLFW_DECORATED to GLFW_TRUE has no effect with glfwSetWindowAttrib()

### ISSUE THREAD

**Issue by manimax3** [glfw/glfw#1566]
OS: OpenSuse Tumbleweed with KDE Plasma on X11
GLFW version: 3.3

Hello,

I can't turn the window decorations back on with glfwSetWindowAttrib(). The other way round i.e. turning them off does work however.
Maybe decorations inherently cant be enabled after the window creation?  

```c
#include <GLFW/glfw3.h>

int main(void)
{
    GLFWwindow *window;

    if (!glfwInit())
        return -1;

    glfwWindowHint(GLFW_DECORATED, GLFW_FALSE);
    window = glfwCreateWindow(640, 480, "Hello World", NULL, NULL);

    if (!window) {
        glfwTerminate();
        return -1;
    }

    glfwMakeContextCurrent(window);

    glfwSetWindowAttrib(window, GLFW_DECORATED, GLFW_TRUE); // Has no effect on the window

    while (!glfwWindowShouldClose(window)) {
        glClear(GL_COLOR_BUFFER_BIT);
        glfwSwapBuffers(window);
        glfwPollEvents();
    }

    glfwTerminate();
    return 0;
}
```

**Comment by elmindreda** [comment]
Thank you for reporting this! Please test if the `x11-fix-enable-decorations` branch if possible and see if it fixes this issue for you.

**Comment by manimax3** [comment]
Just tried and it does not seem to have fixed the issue for me.

**Comment by elmindreda** [comment]
Okay, thank you for testing!

**Comment by elmindreda** [comment]
Please test if the update to the `x11-fix-enable-decorations` branch works better. It appears to work on my Kubuntu VM.

**Comment by manimax3** [comment]
Yes, works fine for me. Thank you very much.

**Comment by elmindreda** [comment]
Merged as 5fc4c01302b0729fd9f39e8c504e28cf7347153c. Thank you for reporting and testing!
